force-app/main/default/classes/CountTasksTest.cls

```diff
@@ -0,0 +1,183 @@
+/**
+ * Apex Test Class for all scenarios
+ * (insert, update, delete, close task,
+ * re open task, reparenting task)
+ */
+@isTest
+public class CountTasksTest {
+  private static final Integer ACCOUNTS_NUMBER = 5;
+  private static final Integer TASKS_NUMBER = 5;
+
+  @TestSetup
+  private static void makeData() {
+    List<Account> accounts = new List<Account>();
+
+    Id recordTypeIdUS = Schema.SObjectType.Account.getRecordTypeInfosByDeveloperName()
+      .get('United_States')
+      .getRecordTypeId();
+
+    for (Integer i = 0; i < ACCOUNTS_NUMBER; i++) {
+      Account acc = new Account(
+        Name = 'Test Account ' + i,
+        Type = 'Prospect',
+        RecordTypeId = recordTypeIdUS
+      );
+      accounts.add(acc);
+    }
+
+    insert accounts;
+
+    List<Task> tasks = new List<Task>();
+    for (Account acc : accounts) {
+      for (Integer i = 0; i < TASKS_NUMBER; i++) {
+        Task tskCall = new Task(
+          Subject = 'Call',
+          Type = 'Call',
+          WhatId = acc.Id
+        );
+        tasks.add(tskCall);
+        Task tskEmail = new Task(
+          Subject = 'Email',
+          Type = 'Email',
+          WhatId = acc.Id
+        );
+        tasks.add(tskEmail);
+        Task tskMeeting = new Task(
+          Subject = 'Meeting',
+          Type = 'Meeting',
+          WhatId = acc.Id
+        );
+        tasks.add(tskMeeting);
+        Task tskOther = new Task(
+          Subject = 'Other',
+          Type = 'Other',
+          WhatId = acc.Id
+        );
+        tasks.add(tskOther);
+      }
+    }
+    insert tasks;
+  }
+
+  @isTest
+  private static void shouldHaveAccountsAndTasksInDatabase() {
+    Assert.areEqual(
+      [SELECT COUNT() FROM Account],
+      ACCOUNTS_NUMBER,
+      'Failed to get expected amount of Accounts'
+    );
+    Integer numberOfTaskTypes = 4;
+    Assert.areEqual(
+      [SELECT COUNT() FROM Task],
```

Comment:
My bad, the arguments are in the wrong order.

[areEqual(expected, actual, msg)](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_class_System_Assert.htm#apex_System_Assert_areEqual)


expected comes first, actual second. Otherwise the error messages look wrong. 
Asserts that the first two arguments are the same.